asciidoctorj-core/src/main/java/org/asciidoctor/jruby/ast/impl/FootnoteImpl.java

```diff
@@ -0,0 +1,45 @@
+package org.asciidoctor.jruby.ast.impl;
+
+import org.jruby.RubyStruct;
+
+import org.asciidoctor.ast.Footnote;
+import org.jruby.javasupport.JavaEmbedUtils;
+
+public class FootnoteImpl implements Footnote {
+
+    private static final String INDEX_KEY_NAME = "index";
+    private static final String ID_KEY_NAME = "id";
+    private static final String TEXT_KEY_NAME = "text";
+
+    private Long index;
+    private String id;
+    private String text;
+
+    private static Object aref(RubyStruct s, String key)  {
+       return JavaEmbedUtils.rubyToJava(s.aref(JavaEmbedUtils.javaToRuby(s.getRuntime(), key)));
```

Comment:
As we know that we need a RubyString would it make sense to directly create it instead of calling `JavaEmbedUtils.javaToRuby(s.getRuntime(), key)`?

Reply:
Oh! Maybe! I think you are suggesting the constants could be RubyStrings?  And the aref signature would be `aref(RubyStruct s, RubyString key)`.  If so, ya, I like it!

Reply:
I could stay `private static Object aref(RubyStruct s, String key)`, but instead of calling `JavaEmbedUtils.javaToRuby(s.getRuntime(), key)`to get the key, we could directly create the RubyString from the correct RubyRuntime with `s.getRuntime().newString(key)`.
Because I guess we should be careful to make sure that all objects belong to the same Ruby instance.

Reply:
Oh right... coolio! Will do!